apps.go

```diff
@@ -0,0 +1,278 @@
+	ID                   string      `json:"id"`
+	Spec                 *AppSpec    `json:"spec"`
+	DefaultIngress       string      `json:"default_ingress"`
+	CreatedAt            time.Time   `json:"created_at"`
+	UpdatedAt            time.Time   `json:"updated_at,omitempty"`
+	ActiveDeployment     *Deployment `json:"active_deployment,omitempty"`
+	InProgressDeployment *Deployment `json:"in_progress_deployment,omitempty"`
+}
+
+// Deployment represents a deployment for an app.
+type Deployment struct {
+	ID          string                  `json:"id"`
+	Spec        *AppSpec                `json:"spec"`
+	Services    []*DeploymentService    `json:"services,omitempty"`
+	Workers     []*DeploymentWorker     `json:"workers,omitempty"`
+	StaticSites []*DeploymentStaticSite `json:"static_sites,omitempty"`
+
+	Cause    string              `json:"cause"`
+	Progress *DeploymentProgress `json:"progress"`
+
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
+
+// DeploymentService represents a service component in a deployment.
+type DeploymentService struct {
+	Name             string `json:"name,omitempty"`
+	SourceCommitHash string `json:"source_commit_hash"`
+}
+
+// DeploymentWorker represents a worker component in a deployment.
+type DeploymentWorker struct {
+	Name             string `json:"name,omitempty"`
+	SourceCommitHash string `json:"source_commit_hash"`
+}
+
+// DeploymentStaticSite represents a static site component in a deployment.
+type DeploymentStaticSite struct {
+	Name             string `json:"name,omitempty"`
+	SourceCommitHash string `json:"source_commit_hash"`
+}
+
+// DeploymentProgress represents the total progress of a deployment.
+type DeploymentProgress struct {
+	PendingSteps int `json:"pending_steps"`
+	RunningSteps int `json:"running_steps"`
+	SuccessSteps int `json:"success_steps"`
+	ErrorSteps   int `json:"error_steps"`
+	TotalSteps   int `json:"total_steps"`
+
+	Steps []*DeploymentProgressStep `json:"steps"`
+}
+
+// DeploymentProgressStep represents the progress of a deployment step.
+type DeploymentProgressStep struct {
+	Name      string                    `json:"name"`
+	Status    string                    `json:"status"`
+	Steps     []*DeploymentProgressStep `json:"steps,omitempty"`
+	Attempts  uint32                    `json:"attempts"`
+	StartedAt time.Time                 `json:"started_at,omitempty"`
+	EndedAt   time.Time                 `json:"ended_at,omitempty"`
+}
+
+// AppLogs represent app logs.
+type AppLogs struct {
+	LiveURL      string   `json:"live_url"`
+	HistoricURLs []string `json:"historic_urls"`
+}
+
+// AppCreateRequest represents a request to create an app.
+type AppCreateRequest struct {
+	Spec *AppSpec `json:"spec"`
+}
+
+// AppUpdateRequest represents a request to update an app.
+type AppUpdateRequest struct {
+	Spec *AppSpec `json:"spec"`
+}
+
+type appRoot struct {
+	App *App `json:"app"`
+}
+
+type appsRoot struct {
+	Apps []*App `json:"apps"`
```

Comment:
Are these responses paginated? If so, `Links` and `Meta` should be included here as well.